packages/connection-form/src/utils/csfle-handler.ts

```diff
@@ -0,0 +1,222 @@
+  };
+  if (!action.value) {
+    delete kms[action.key];
+  } else {
+    kms[action.key] = action.value;
+  }
+  if (Object.keys(kms).length === 0) {
+    kms = undefined;
+  }
+  return {
+    connectionOptions: {
+      ...connectionOptions,
+      fleOptions: {
+        storeCredentials: false,
+        ...connectionOptions.fleOptions,
+        autoEncryption: {
+          ...autoEncryption,
+          kmsProviders: {
+            ...autoEncryption.kmsProviders,
+            [action.kmsProvider]: kms,
+          },
+        },
+      },
+    },
+  };
+}
+
+export function handleUpdateCsfleKmsTlsParam({
+  action,
+  connectionOptions,
+}: {
+  action: UpdateCsfleKmsTlsAction;
+  connectionOptions: ConnectionOptions;
+}): {
+  connectionOptions: ConnectionOptions;
+} {
+  connectionOptions = cloneDeep(connectionOptions);
+  const autoEncryption = connectionOptions.fleOptions?.autoEncryption ?? {};
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  let tls: any = { ...(autoEncryption.tlsOptions?.[action.kmsProvider] ?? {}) };
+  if (!action.value) {
+    delete tls[action.key];
+  } else {
+    tls[action.key] = action.value;
+  }
+  if (Object.keys(tls).length === 0) {
+    tls = undefined;
+  }
+  return {
+    connectionOptions: {
+      ...connectionOptions,
+      fleOptions: {
+        storeCredentials: false,
+        ...connectionOptions.fleOptions,
+        autoEncryption: {
+          ...autoEncryption,
+          tlsOptions: {
+            ...autoEncryption.tlsOptions,
+            [action.kmsProvider]: tls,
+          },
+        },
+      },
+    },
+  };
+}
+
+export function hasAnyCsfleOption(o: Readonly<AutoEncryptionOptions>): boolean {
+  return !!(
+    o.bypassAutoEncryption ||
+    o.keyVaultNamespace ||
+    o.schemaMap /* TODO(COMPASS-5645): encryptedFieldConfig */ ||
+    [
+      ...Object.values(o.tlsOptions ?? {}),
+      ...Object.values(o.kmsProviders ?? {}),
+    ]
+      .flatMap((o) => Object.values(o))
+      .filter(Boolean).length > 0
+  );
+}
+
+export function textToEncryptedFieldConfig(text: string): Document | undefined {
+  if (!text.trim()) {
+    return undefined;
+  }
+
+  // We use `$compass`-prefixed strings here since the keys of
+  // `encryptedFieldConfigMap` refer to databases and
+  // those never contain `$`.
+
+  let parsed: Document = {};
+  try {
+    parsed = queryParser(text);
+    if (!parsed || typeof parsed !== 'object') {
+      // We've hit the condition in
+      // https://github.com/mongodb-js/ejson-shell-parser/blob/c9c0145ababae52536ccd2244ac2ad01a4bbdef3/src/index.ts#L36
+      // in which instead of returning a parsed value or throwing an error,
+      // the query parser just returns an empty string when encountering
+      // input that can be parsed as JS but not as a valid query.
+      // Unfortunately, this also means that all context around what
+      // caused this error is unavailable here.
```

Comment:
Should we ticket to improve this? I guess it's maybe tech debt, but could be a feature for this project if we want. No worries if now is not the time.

Reply:
Yes :+1: Opened COMPASS-5689 for this.